www/js/copy.js

```diff
@@ -0,0 +1,176 @@
+const { store } = require('electron').remote.require('./app');
+
+// other essential imports
+const copy = require('copy-to-clipboard');
+const sanitizeHtml = require('sanitize-html');
+const anvaad = require('anvaad-js');
+const search = require('./search');
+const baniDB = require('../js/banidb/index.js');
+
+const baniLengthMap = search.baniLengthCols;
+
+// defualt these to true, so that we only need to change them if they are turned off
+let copyEngTranslation = true;
+let copyPunjabiTranslation = true;
+let copyTranslit = true;
+
+const baniLength = store.getUserPref('toolbar.gurbani.bani-length');
+
+// storing pulled shabad/bani from db in array to avoid calling db too much
+const pankteeArray = [];
+
+function findLinePosition() {
+  const navParent = document.querySelector('#shabad.gurmukhi');
+  const currentPanktee = document.querySelector('a.panktee.seen_check.current');
+  const arr = Array.from(navParent.childNodes);
+  const linePos = arr.indexOf(currentPanktee.parentNode);
+  return linePos;
+}
+/**
+ *
+ * @param {String} htmlString explanation text from ceremonies
+ * uses sanitize html to remove all html tags from text
+ */
+function stripHTML(htmlString) {
+  const allowedTags = [];
+  const allowedAttributes = {};
+  return sanitizeHtml(htmlString, { allowedTags, allowedAttributes });
+}
+/**
+ *
+ * @param {Objects} unmapped rows from the DB
+ * remaps each row
+ */
+function remapShabad(unmapped) {
+  pankteeArray.length = 0;
+  unmapped.forEach(row => {
+    const remappedRow = global.controller.remapLine(row);
+    pankteeArray.push(remappedRow);
+  });
+}
+/**
+ *
+ * @param {Objects} unmapped rows fresh from DB
+ * for each row, it sees if row.Verse (normal panktees) row.Custom (custom symbols or text) is available
+ * if not, it just displays title (because those two will only not be available at the position of the title)
+ * Then, it remapps the correct part of the row Object
+ */
+function remapBani(unmapped) {
```

Comment:
@tsingh777 sukhmani sahib has an extra row.Verse at the beginning that offsets the rest of the bani by 1 (its only for sukhmani sahib). Whats the best way to fix that?